opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -161,56 +261,40 @@ def force_flush(self) -> bool:
 
         return metric_reader_result and metric_exporter_result
 
-    def register_metric_reader(self, metric_reader: "MetricReader") -> None:
-        # FIXME protect this method against race conditions
-        self._metric_readers.append(metric_reader)
-
-    def register_metric_exporter(
-        self, metric_exporter: "MetricExporter"
-    ) -> None:
-        # FIXME protect this method against race conditions
-        self._metric_exporters.append(metric_exporter)
-
-    def register_view(self, view: "View") -> None:
-        # FIXME protect this method against race conditions
-        self._views.append(view)
-
-
-class MetricReader(ABC):
-    def __init__(self):
-        self._shutdown = False
-
-    @abstractmethod
-    def collect(self):
-        pass
-
     def shutdown(self):
-        # FIXME this will need a Once wrapper
-        self._shutdown = True
+        # FIXME implement a timeout
+
+        if self._shutdown:
+            _logger.warning("shutdown can only be called once")
+            return False
 
+        result = True
 
-class MetricExporter(ABC):
-    def __init__(self):
-        self._shutdown = False
+        for metric_reader in self._metric_readers:
+            result = result and metric_reader.shutdown()
 
-    @abstractmethod
-    def export(self):
-        pass
+        for metric_exporter in self._metric_exporters:
+            result = result and metric_exporter.shutdown()
 
-    def shutdown(self):
-        # FIXME this will need a Once wrapper
         self._shutdown = True
 
+        if self._atexit_handler is not None:
+            unregister(self._atexit_handler)
+            self._atexit_handler = None
 
-class View:
-    pass
-
+        return result
 
-class ConsoleMetricExporter(MetricExporter):
-    def export(self):
-        pass
+    def get_meter(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        schema_url: Optional[str] = None,
+    ) -> Meter:
 
+        if self._shutdown:
+            _logger.warning(
+                "A shutdown `MeterProvider` can not provide a `Meter`"
+            )
+            return _DefaultMeter(name, version=version, schema_url=schema_url)
 
-class SDKMetricReader(MetricReader):
-    def collect(self):
-        pass
+        return Meter(InstrumentationInfo(name, version, schema_url), self)
```

Comment:
nit you can share a single instance of `InstrumentationInfo` with all meters as it's immutable, rather than create it repeatedly

Reply:
`InstrumentationInfo` may be immutable but I don't think that is relevant here, we need to create a new one per `Meter`, this that is being done here which is the same thing as we do in [tracing](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L1116).

As far as I know, `MeterProvider.get_meter` would receive `name`, `version` and `schema_url`, arguments that would be used to create a new `InstrumentationInfo` object to pass to the new `Meter` instance. If it is how you say, then we should not have the `name`, `version` and `schema_url` as arguments of `MeterProvider.get_meter` but as arguments of `MeterProvider.__init__`.

Reply:
Oops sorry, I got confused thought this was `Meter` passing off to. This looks right to me